src/go/rdctl/cmd/start.go

```diff
@@ -0,0 +1,178 @@
+	rootCmd.AddCommand(startCmd)
+	updateCommonStartAndSetCommands(startCmd)
+	startCmd.Flags().StringVarP(&applicationPath, "path", "p", "", "Path to main executable.")
+}
+
+// Unavoidable race condition here.
+// There's no system-wide mutex that will let us guarantee that if rancher desktop is running when
+// we test it (easiest to just try to get the settings), that it will still be running when we
+// try to upload the settings (if any were specified).
+func doStartOrSetCommand(cmd *cobra.Command) error {
+	_, err := getListSettings()
+	if err == nil {
+		if applicationPath != "" {
+			// `--path | -p` is not a valid option for `rdctl set...`
+			return fmt.Errorf("--path specified but Rancher Desktop is already running", applicationPath)
+		}
+		return doSetCommand(cmd)
+	}
+	return doStartCommand(cmd)
+}
+
+func doStartCommand(cmd *cobra.Command) error {
+	var commandLineArgs []string
+
+	if cmd.Flags().Changed("container-engine") {
+		commandLineArgs = append(commandLineArgs, "--kubernetes-container-engine", specifiedSettings.ContainerEngine)
+	}
+	if cmd.Flags().Changed("kubernetes-enabled") {
+		commandLineArgs = append(commandLineArgs, "--kubernetes-enabled", strconv.FormatBool(specifiedSettings.Enabled))
+	}
+	if cmd.Flags().Changed("kubernetes-version") {
+		commandLineArgs = append(commandLineArgs, "--kubernetes-version", specifiedSettings.Version)
+	}
+	if cmd.Flags().Changed("flannel-enabled") {
+		commandLineArgs = append(commandLineArgs, "--kubernetes-options-flannel", strconv.FormatBool(specifiedSettings.Flannel))
+	}
+	if applicationPath == "" {
+		pathLookupFuncs := map[string]func() string{
+			"windows": getWindowsRDPath,
+			"linux":   getLinuxRDPath,
+			"darwin":  getMacOSRDPath,
+		}
+		getPathFunc, ok := pathLookupFuncs[runtime.GOOS]
+		if !ok {
+			return fmt.Errorf("don't know how to find the path to Rancher Desktop on OS %s", runtime.GOOS)
+		}
+		applicationPath = getPathFunc()
+		if applicationPath == "" {
+			return fmt.Errorf("no executable found in the default location; please retry with the --path|-p option")
+		}
+	}
+	return launchApp(applicationPath, commandLineArgs)
+}
+
+func launchApp(applicationPath string, commandLineArgs []string) error {
+	var commandName string
+	var args []string
+
+	if runtime.GOOS == "darwin" {
+		commandName = "open"
+		args = append(args, "-a", applicationPath)
+		if len(commandLineArgs) > 0 {
+			args = append(args, "--args")
+			args = append(args, commandLineArgs...)
+		}
+	} else {
+		commandName = applicationPath
+		args = commandLineArgs
+	}
+	// Include this output because there's a delay before the UI comes up.
+	// Without this line, it might look like the command doesn't work.
+	fmt.Fprintf(os.Stderr, "About to launch %s %s ...\n", commandName, strings.Join(args, " "))
+	cmd := exec.Command(commandName, args...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	return cmd.Run()
+}
+
+func getWindowsRDPath() string {
+	localAppDataDir := os.Getenv("LOCALAPPDATA")
```

Comment:
Ideally we'd do this via [`SHGetKnownFolderPath`](https://docs.microsoft.com/en-us/windows/win32/api/shlobj_core/nf-shlobj_core-shgetknownfolderpath) (with `FOLDERID_UserProgramFiles`) instead; the environment variables can be misconfigured. But I guess this is good enough for the initial implementation…

Reply:
If it's good enough, let's leave it for now.